flows/common/counter.go

```diff
@@ -203,17 +187,24 @@ func (f *counters) rxRate() float32 {
 	return float32(sum) / delta * 8.0
 }
 
-// telemetry generates the set of gNMI Notifications that describes the flow's current state.
-// The target argument specifies the Target value that should be included in the notifications.
-func (f *counters) telemetry(target string) []*gpb.Notification {
-	type datapoint struct {
-		d  *otgyang.Device
-		ts int64
-	}
+type datapoint struct {
+	d  *otgyang.Device
+	ts int64
+}
+
+// datapoints returns the set of telemetry updates with timestamps that need to be sent for this flow.
+//
+// TODO(robjs): with sufficient numbers of flows, then this function ends up holding the f.mu lock regularly
+// and causing lock contention with packets that are being sent and received. An alternate approach to either
+// batch updates from the Tx/Rx goroutines, or push updates from the Tx/Rx goroutines is needed to remove the
```

Comment:
nit: as

Reply:
I think that `is` is right here -- i.e., an approach to do A or B is needed. However, the sentence was a bit runaway so I restructured it.